utils/globalvar_test.go

```diff
@@ -0,0 +1,53 @@
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseIntEnvVar(t *testing.T) {
+	tests := []struct {
+		name         string
+		varName      string
+		varValue     string
+		defaultValue int
+		expected     int
+		expectError  bool
+	}{
+		{
+			name:         "environment variable not set",
+			varName:      "TEST_ENV_VAR_NOT_SET",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with valid integer",
+			varName:      "TEST_ENV_VAR_VALID",
+			varValue:     "20",
+			defaultValue: 10,
+			expected:     20,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with invalid integer",
+			varName:      "TEST_ENV_VAR_INVALID",
+			varValue:     "invalid",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Setenv(tt.varName, tt.varValue)
+			defer t.Setenv(tt.varName, "")
```

Comment:
nitpick(non-blocking): there’s no need to use defer `testing.T.Setenv` rolls back your change at the end of test automatically